contrib/platform/src/com/sun/jna/platform/win32/Advapi32.java

```diff
@@ -284,6 +301,292 @@ boolean CreateWellKnownSid(int wellKnownSidType, PSID domainSid,
+     *          ACL_REVISION_DS. Use ACL_REVISION_DS if the ACL contains object-specific ACEs.
+     * @param AccessMask
+     *          Specifies the mask of access rights to be granted to the specified SID.
+     * @param pSid
+     *          A pointer to the SID representing a user, group, or logon account being granted access.
+     * @return If the function succeeds, the return value is nonzero. If the
+     *         function fails, the return value is zero. For extended error
+     *         information, call GetLastError.
+     */
+    boolean AddAccessAllowedAce(ACL pAcl, int dwAceRevision, int AccessMask, PSID pSid);
+
+    /**
+     * The AddAce function adds one or more access control entries (ACEs) to a specified access control list (ACL).
+     * @param pAcl
+     *          A pointer to an ACL. This function adds an access-allowed ACE to the end of this ACL.
+     *          The ACE is in the form of an ACCESS_ALLOWED_ACE structure.
+     * @param dwAceRevision
+     *          Specifies the revision level of the ACL being modified. This value can be ACL_REVISION or
+     *          ACL_REVISION_DS. Use ACL_REVISION_DS if the ACL contains object-specific ACEs.
+     * @param AceFlags
+     *          A set of bit flags that control ACE inheritance. The function sets these flags in the AceFlags
+     *          member of the ACE_HEADER structure of the new ACE. This parameter can be a combination
+     *          of the following values: CONTAINER_INHERIT_ACE, INHERIT_ONLY_ACE, INHERITED_ACE,
+     *          NO_PROPAGATE_INHERIT_ACE, and OBJECT_INHERIT_ACE
+     * @param AccessMask
+     *          Specifies the mask of access rights to be granted to the specified SID.
+     * @param pSid
+     *          A pointer to the SID representing a user, group, or logon account being granted access.
+     * @return If the function succeeds, the return value is nonzero. If the
+     *         function fails, the return value is zero. For extended error
+     *         information, call GetLastError.
+     */
+    boolean AddAccessAllowedAceEx(ACL pAcl, int dwAceRevision, int AceFlags, int AccessMask, PSID pSid);
+
+    /**
+     * The GetAce function obtains a pointer to an access control entry (ACE) in an access
+     * control list (ACL).
+     * @param pAcl
+     *          A pointer to an ACL that contains the ACE to be retrieved.
+     * @param dwAceIndex
+     *          The index of the ACE to be retrieved. A value of zero corresponds to the first ACE in
+     *          the ACL, a value of one to the second ACE, and so on.
+     * @param pAce
+     *          A pointer to a pointer that the function sets to the address of the ACE.
+     * @return If the function succeeds, the return value is nonzero. If the
+     *         function fails, the return value is zero. For extended error
+     *         information, call GetLastError.
+     */
+    boolean GetAce(ACL pAcl, int dwAceIndex, PointerByReference pAce);
```

Comment:
That structure is abstract, so while we could define the function to take it, users would need to use their own implementation or pass in a ACCESS_ALLOWED_ACE or ACCESS_DENIED_ACE instead. I did try this, passing an instance of ACCESS_ALLOWED_ACE in, but now the new write() overload crashes from autoWrite because the psid member is null. Actually, given this find, I should probably do something about null pointer.